agent/agent.go

```diff
@@ -421,22 +421,24 @@ func (a *Agent) Publisher(ctx context.Context, subscriptionID string) (exec.LogP
 		return err
 	})
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
 	return exec.LogPublisherFunc(func(ctx context.Context, message api.LogMessage) error {
-		select {
-		case <-ctx.Done():
-			client.CloseSend()
-			return ctx.Err()
-		default:
-		}
+			select {
+			case <-ctx.Done():
+				client.CloseSend()
+				return ctx.Err()
+			default:
+			}
 
-		return client.Send(&api.PublishLogsMessage{
-			SubscriptionID: subscriptionID,
-			Messages:       []api.LogMessage{message},
-		})
-	}), nil
+			return client.Send(&api.PublishLogsMessage{
+				SubscriptionID: subscriptionID,
+				Messages:       []api.LogMessage{message},
+			})
+		}), func() {
+			client.CloseSend()
```

Comment:
This is not new in this PR, but I just noticed that `client` is a misleading name:

```
client, err = api.NewLogBrokerClient(session.conn).PublishLogs(ctx)
```

It's really a stream, not a client.